docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -29,30 +30,85 @@ Include it in your module file like this:
     # ... snip ...
     '''
 
-The ``description``, and ``notes`` fields
-support formatting with some special macros.
 
-These formatting functions are ``U()``, ``M()``, ``I()``, and ``C()``
-for URL, module, italic, and constant-width respectively. It is suggested
-to use ``C()`` for file and option names, and ``I()`` when referencing
-parameters; module names should be specified as ``M(module)``.
+The following fields can be used and are all required unless specified otherwise
+
+* ``module:``
+  The name of the module, same as the file without `.py`.
+* ``short_description:``
+
+  * Displayed on the :doc:`../list_of_all_modules` page and ``ansible-doc -l``.
+  * Must have enough detail without the context of the category.
+  * Without a trailing full stop
+* ``description:``
+  * A longer description than `short_description`, which goes into more detail, generally two or more sentences.
+  * Must be written in full sentences, i.e. with capital letters and fullstops.
+  * Shouldn't mention the name module.
+* ``version_added:``
+  The version of Ansible when the module was added.
+  This is a `string`, and not a float, i.e. ``version_added: "2.1"``
```

Comment:
I wonder if we want a pull box somewhere that reiterates the string rather than float message with some rationale?

Maybe soemthing like:
```
.. note:: Versions should be strings

    In general, care should be taken to list versions as strings rather than floats.  This has
    several reasons:
    * Versions may not always be floats, for instance "2.1.0" or "2.0alpha1".
    * Versions may not compare correctly if used as a float.  A float will tempt
       naive comparison between 2.3 and 2.12 which won't generate the correct
       ordering.  Strings "2.3" and "2.12" require a bit of code which will let the
       programmer do it correctly.